src/test/java/org/icatproject/ids/integration/util/client/TestingClient.java

```diff
@@ -911,4 +927,21 @@ public void write(String sessionId, DataSelection data, Integer sc) throws NotIm
         }
     }
 
+    private void checkResponseConformity(HttpResponse response) throws InternalException {
+        
+        StatusLine status = response.getStatusLine();
+        var statusCode = status.getStatusCode();
+        if(statusCode == 200 && response.getEntity() != null) {
+            //we have a status of 200 and a payload. Check for header conformity
+            Boolean chunked = response.getFirstHeader("Transfer-Encoding") != null ? response.getFirstHeader("Transfer-Encoding").getValue().contains("chunked")
+                                                                                        : false;
+            
+            if ( (response.getFirstHeader("Content-Length") == null && !chunked)
+                || (response.getFirstHeader("Content-Length") != null && chunked ) ) {
```

Comment:
Haven't used assert here to be able to provide a helpful text message.

Reply:
Agree. Assert should rather be used in tests. In helper classes like this one it's better to raise an exception if something goes wrong.